### PR TITLE
Event controller

### PR DESCRIPTION
**Pull request by shinwachi**
Added Event Creation API with method and user ID checks (other fields are incomplete, but out of scope of this task).  Added User entity with database seeder.  Added custom error message for cases where IDs don't relate to either user or methods.

### CODE REVIEW

File: src/main/java/edu/harvard/h2ms/web/controller/EventController.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+		
+		Event event = new Event();
+		
+		// parse time
+		// https://stackoverflow.com/questions/22463062/how-to-parse-format-dates-with-localdatetime-java-8
+		CharSequence dateTime = (CharSequence)payload.get("date-time");
+		DateTimeFormatter f = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss'Z'");
+		LocalDateTime dateTime2 = LocalDateTime.from(f.parse(dateTime));
+		// All this work for just this.
+		event.setTimestamp(new Time(dateTime2.toEpochSecond(ZoneOffset.ofHours(0))));
```
Comment (stbenjam) [inline src/main/java/edu/harvard/h2ms/web/controller/EventController.java]: I don't think you really need to do this, Spring creates API end points for you based on the Event model for free, including read and create.   Visit http://localhost:8090/swagger-ui.html on the master branch, which automatically generates documentation for the API end points we already have. You can see we have create, delete, edit, and all kinds of end points already for event.

That's why I don't really even think we need two separate tasks for "READ" and "CREATE".  The first step should really be to get the Event model correct, right now it's not.

For example, we are indeed missing the dateTime field (among others). The only thing needed is to edit the Event more in domain/core, and add the fields we need.

Comment (stbenjam) [reply src/main/java/edu/harvard/h2ms/web/controller/EventController.java]: I'm not sure we should use java.sql.Time, if you use java.date.LocalDateTime, the format will be more natural and like what you have here.

Likewise for the subject and observer, they can just be User objects.  We might need some helper methods to find things, but I think the API handles this already.

I don't know how people set it up in the real world, we should get @MichaelDelaney to clarify.

Comment (shinwachi) [reply src/main/java/edu/harvard/h2ms/web/controller/EventController.java]: I got the controller based on existing Event model.  I was getting null objects in the fields when I did the autowire magic, so I cooked up the data entry functions.  Do you think I should I cancel this commit until method model is clear(er)?

**Comment by shinwachi** [comment]
Looks like further modification is needed in the data model.  Be back after.

**Pull request closed without merging**